pkg/cli/publish.go

```diff
@@ -71,17 +78,53 @@ func PublishCmd(ctx context.Context, outputRefs string, opts ...build.Option) er
 		return err
 	}
 
+	if len(archs) == 0 {
+		archs = types.AllArchs
+	}
+	if len(bc.ImageConfiguration.Archs) == 0 {
+		bc.ImageConfiguration.Archs = archs
+	}
+
 	log.Printf("building tags %v", bc.Tags)
 
-	layerTarGZ, err := bc.BuildLayer()
-	if err != nil {
-		return fmt.Errorf("failed to build layer image: %w", err)
-	}
-	defer os.Remove(layerTarGZ)
+	var digest name.Digest
+	switch len(bc.ImageConfiguration.Archs) {
+	case 0:
+		return fmt.Errorf("no archs requested: %w", err)
```

Comment:
What error is being wrapped here?

Reply:
Good catch! Fixed. Technically I think it should never even happen here, but just in case.